buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/fixtures/RunEmbeddedGradle.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.gradlebuild.test.fixtures
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.internal.ProcessOperations
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Classpath
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.options.Option
+import java.io.File
+
+
+abstract class RunEmbeddedGradle : DefaultTask() {
+
+    @get:Classpath
+    abstract val gradleClasspath: ConfigurableFileCollection
+
+    @get:Input
+    @set:Option(option = "arg", description = "Arguments for invoking Gradle")
+    var args: List<String> = listOf()
```

Comment:
It would be better if we can pass all args one-time: `./gradlew fileWatching:runDevGradle --working-dir "/home/user/multiproject" --args 'task1 task2 task3 -Dorg.gradle.debug=true'`, but it's good enough now.

Reply:
I added support for using `--args` and a space separated list of arguments.